game/gui/map_utils/map_drawer.py

```diff
@@ -0,0 +1,118 @@
+import pygame
+from pygame import Surface
+from pygame.sprite import Sprite
+
+from constants import SCREEN_WIDTH, SCREEN_HEIGHT
+from entity.map_features.base import Base
+from entity.map_features.empty import Empty
+from entity.map_features.obstacle import Obstacle
+from entity.tanks.tank import Tank
+from gui.map_utils.explosion import Explosion
+from gui.map_utils.projectile import Projectile
+from gui.map_utils.scoreboard import Scoreboard
+from gui.map_utils.tank_drawer import TankDrawer
+from map.hex import Hex
+
+
+class MapDrawer:
+    def __init__(self, map_size: int, players: dict, game_map: dict, current_turn: list[1]):
+        self.__num_of_radii = (map_size - 1) * 2 * 2
+        self.__map_size = map_size
+        self.__turn: list[1] = current_turn
+        self.__max_damage_points: int = 0
+        self.__players = players
+        self.__map = game_map
+
+        Hex.radius_x = SCREEN_WIDTH // self.__num_of_radii  # number of half radii on x axis
+        Hex.radius_y = SCREEN_HEIGHT // self.__num_of_radii  # number of half radii on y axis
+
+        self.__scoreboard = Scoreboard(players)
+        self.__scoreboard.update_image_size(Hex.radius_x * 2, Hex.radius_y * 2)
+        self.__scoreboard.set_radii(Hex.radius_x / 3, Hex.radius_y / 3)
+        self.__font_size = round(1.2 * min(Hex.radius_y, Hex.radius_x))
+        self.__font = None
+        self.__explosion_group = pygame.sprite.Group()
+        self.__projectile_group = pygame.sprite.Group()
+        # note: this could be moved somewhere else
+        self.__explosion_delay = Projectile.get_travel_time()
+
+        # map legend
+        self.__map_legend_items = []
+        features = [Empty, Base, Obstacle]
+        # this will always be on the screen and on top right
+        x, y, z = self.__map_size + 1, self.__map_size - 2, -self.__map_size - 1
+        for i, feature in enumerate(features):
+            self.__map_legend_items.append(feature((x, y - i, z + i)))
+
+        # tanks
+        self.__tanks = pygame.sprite.Group()
+        for _, entities in self.__map.items():
+            tank = entities['tank']
+            if tank is not None:
+                self.__tanks.add(TankDrawer(tank))
+
+    def draw(self, screen: Surface):
+        if self.__font is None:
+            self.__font = pygame.font.SysFont('georgia', self.__font_size, bold=True)
+
+        # fill with background color
+        screen.fill((47, 31, 128))
```

Comment:
Magic numbers. Named constant would be a better option.

Reply:
Hello Vladislav,

this will be changed